source/mesh_deformation/interface.cc

```diff
@@ -464,6 +489,127 @@ namespace aspect
 
 
 
+    template <int dim>
+    AffineConstraints<double> MeshDeformationHandler<dim>::make_initial_constraints()
+    {
+      AssertThrow(this->get_parameters().mesh_deformation_enabled, ExcInternalError());
+
+      // initial_deformation_constraints can use the same hanging node
+      // information that was used for mesh_vertex constraints.
+      AffineConstraints<double> initial_deformation_constraints(mesh_locally_relevant);
+      initial_deformation_constraints.merge(mesh_vertex_constraints);
+
+      // Add the vanilla periodic boundary constraints
+      std::set< types::boundary_id > periodic_boundaries;
+      using periodic_boundary_pairs = std::set< std::pair< std::pair<types::boundary_id, types::boundary_id>, unsigned int> >;
+      periodic_boundary_pairs pbp = this->get_geometry_model().get_periodic_boundary_pairs();
+      for (periodic_boundary_pairs::iterator p = pbp.begin(); p != pbp.end(); ++p)
+        {
+          periodic_boundaries.insert((*p).first.first);
+          periodic_boundaries.insert((*p).first.second);
+
+          DoFTools::make_periodicity_constraints(mesh_deformation_dof_handler,
+                                                 (*p).first.first,
+                                                 (*p).first.second,
+                                                 (*p).second,
+                                                 initial_deformation_constraints);
+        }
+
+      const std::set<types::boundary_id> all_boundaries = this->get_geometry_model().get_used_boundary_indicators();
+
+      // Periodic boundaries are also considered tangential mesh movement boundaries
+      std::set<types::boundary_id> all_tangential_boundaries;
+      std::set_union(tangential_mesh_boundary_indicators.begin(),tangential_mesh_boundary_indicators.end(),
```

Comment:
These tangential_mesh_boundary_indicators include `Additional tangential mesh boundary indicators` and `Tangential velocity boundary indicators`. A mesh deformation boundary can also have a tangential Stokes velocity boundary condition set on it. So these should be taken out, otherwise the mesh def BI will have no normal flux BC set.